pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -873,13 +873,10 @@ def register_plugins(cls, plugin_registry: Dict) -> None:
             offload_optimizer_device="nvme",
         )
 
-    @property
-    def checkpoint_io(self) -> CheckpointIO:
-        return self._checkpoint_io
-
-    @checkpoint_io.setter
-    def checkpoint_io(self, plugin: CheckpointIO) -> None:
-        raise MisconfigurationException("DeepSpeed currently does not support custom checkpoint plugins.")
+    @DDPPlugin.checkpoint_io.setter
+    def checkpoint_io(self, plugin: Optional[CheckpointIO]) -> None:
+        if plugin is not None:
+            raise MisconfigurationException("DeepSpeed does not support custom checkpoint plugins.")
```

Comment:
nit: just for consistency with your other changes.
```suggestion
    def checkpoint_io(self, io: Optional[CheckpointIO]) -> None:
        if io is not None:
            raise MisconfigurationException("DeepSpeed does not support custom `CheckpointIO` plugin.")
```